statsmodels/base/model.py

```diff
@@ -474,6 +474,135 @@ def fit(self, start_params=None, method='newton', maxiter=100,
+            variables that should be dropped.
+        start_params : None or array_like
+            starting values for the optimization. `start_params` needs to be
+            given in the original parameter space and are internally
+            transformed.
+        **fit_kwds : keyword arguments
+            fit_kwds are used in the optimization of the transformed model.
+
+        Returns
+        -------
+        results : Results instance
+
+        """
+        # wee need to append index of extra params to keep_index as in NegativeBinomial
+        if hasattr(self, 'k_extra') and self.k_extra > 0:
+            # we cannot change the original, TODO: should we add keep_index_params?
+            keep_index = np.array(keep_index, copy=True)
+            keep_index_p = np.concatenate((keep_index,
+                                           list(range(self.exog.shape[1] + self.k_extra))[-self.k_extra:]))
+        else:
+            keep_index_p = keep_index
+
+        # not all models support start_params, drop if None, hide them in fit_kwds
+        if start_params is not None:
+            fit_kwds['start_params'] = start_params[keep_index_p]
+
+        # build auxiliary model and fit
+        init_kwds = self._get_init_kwds()
+        mod_constr = self.__class__(self.endog, self.exog[:, keep_index], **init_kwds)
+        res_constr = mod_constr.fit(**fit_kwds)
+        #switch name, only need keep_index for params belos
+        keep_index = keep_index_p
+
+        # create dummy results Instance, TODO: wire up properly
+        # TODO: this could be moved into separate private method if needed
+        # discrete L1 fit_regularized doens't reestimate AFAICS
+        #res = self.fit(maxiter=0, method='nm', disp=0,
+        # RLM doesn't have method, disp nor warn_convergence keywords
+        # OLS, WLS swallows extra kwds with **kwargs, but doesn't have method='nm'
+        try:
+            res = self.fit(maxiter=0, disp=0, method='nm',
+                           warn_convergence=False) # we get a wrapper back
+        except (TypeError, ValueError):
+            res = self.fit()
+
+        # Warning: make sure we are not just changing the wrapper instead of results #2400
+        if hasattr(res_constr.model, 'scale'):
+            # GLM problem, see #2399
+            res.model.scale = res._results.scale = res_constr.model.scale
+
+        if hasattr(res_constr, 'mle_retvals'):
+            # not available for not scipy optimization, e.g. glm irls
+            res.mle_retvals['fcall'] = res_constr.mle_retvals.get('fcall', np.nan)
+            res.mle_retvals['iterations'] = res_constr.mle_retvals.get(
+                                                            'iterations', np.nan)
+            res.mle_retvals['converged'] = res_constr.mle_retvals['converged']
+            # overwrite all settings
+            #res.mle_retvals.update(res_constr.mle_retvals)  #not yet
+            res.mle_settings.update(res_constr.mle_settings)
```

Comment:
this implies that `mle_settings['start_params']` is in reduced model parameterization, 
I don't think that's a good idea. It's only used by unit tests so far.